src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Android.cs

```diff
@@ -0,0 +1,473 @@
+                if (name.StartsWith("GMT"))
+                {
+                    return new TimeZoneInfo(id, TimeSpan.FromSeconds(ParseGMTNumericZone(name)), id, name, name, null, disableDaylightSavingTime:true);
+                }
+
+                try
+                {
+                    return _GetTimeZone(id, name);
+                }
+                catch (Exception)
+                {
+                    // Should we return null?
+                    return null;
+                }
+            }
+
+            [DllImport(Interop.Libraries.SystemNative, EntryPoint = "SystemNative_GetDefaultTimeZone")]
+            internal static extern string GetDefaultTimeZone();
+
+            internal static TimeZoneInfo? Local
+            {
+                get
+                {
+                    var id = GetDefaultTimeZone();
+                    if (!string.IsNullOrEmpty(id))
+                    {
+                        return GetTimeZone(id, id);
+                    }
+
+                    // If we can't find a default time zone, return UTC
+                    return Utc;
+                }
+            }
+        }
+    }
+
+    [StructLayout(LayoutKind.Sequential, Pack=1)]
+    internal unsafe struct AndroidTzDataHeader
+    {
+        public fixed byte signature[12];
+        public int indexOffset;
+        public int dataOffset;
+    }
+
+    [StructLayout(LayoutKind.Sequential, Pack=1)]
+    internal unsafe struct AndroidTzDataEntry
+    {
+        public fixed byte id[40];
+        public int byteOffset;
+        public int length;
+        public int rawUtcOffset;
+    }
+
+    /*
+     * Android v4.3 Timezone support infrastructure.
+     *
+     * This is a C# port of libcore.util.ZoneInfoDB:
+     *
+     *    https://android.googlesource.com/platform/libcore/+/master/luni/src/main/java/libcore/util/ZoneInfoDB.java
+     *
+     * This is needed in order to read Android v4.3 tzdata files.
+     *
+     * Android 10+ moved the up-to-date tzdata location to a module updatable via the Google Play Store and the
+     * database location changed (https://source.android.com/devices/architecture/modular-system/runtime#time-zone-data-interactions)
+     * The older locations still exist (at least the `/system/usr/share/zoneinfo` one) but they won't be updated.
+     */
+    internal sealed class AndroidTzData
+    {
+        internal static readonly string[] Paths = new string[] {
+            GetApexTimeDataRoot() + "/etc/tz/tzdata", // Android 10+, TimeData module where the updates land
+            GetApexRuntimeRoot() + "/etc/tz/tzdata",  // Android 10+, Fallback location if the above isn't found or corrupted
+            Environment.GetEnvironmentVariable("ANDROID_DATA") + "/misc/zoneinfo/tzdata",
+            Environment.GetEnvironmentVariable("ANDROID_ROOT") + "/usr/share/zoneinfo/tzdata",
+        };
+
+        private string tzdataPath;
+        private Stream? data;
+
+        private string[]? ids;
```

Comment:
Why is it nullable if it's always set?

Reply:
If an exception is thrown before `ReadIndex` is complete, such as `ReadAt` exception throwing, then `ids` doesn't get set.